x-pack/plugins/spaces/public/plugin.tsx

```diff
@@ -60,6 +60,17 @@ export class SpacesPlugin implements Plugin<SpacesPluginSetup, SpacesPluginStart
 
   public setup(core: CoreSetup<PluginsStart, SpacesPluginStart>, plugins: PluginsSetup) {
     const hasOnlyDefaultSpace = this.config.maxSpaces === 1;
+    const onCloud = plugins.cloud !== undefined && plugins.cloud.isCloudEnabled;
+
+    // We only allow "solution" to be set on cloud environments, not on prem
+    // unless the forceSolutionVisibility flag is set
+    let allowSolutionVisibility = this.config.allowSolutionVisibility;
+    if (this.config.forceSolutionVisibility) {
+      allowSolutionVisibility = true;
+    } else if (!onCloud || this.isServerless) {
+      allowSolutionVisibility = false;
+    }
+
```

Comment:
__nit:__ same suggestion as in server's `plugin.ts`.

Reply:
Addressed in https://github.com/elastic/kibana/pull/191743/commits/37fbf7b2e74f3f531a88ffbd54bbeecd4517b276